pkg/crc/config/validations.go

```diff
@@ -103,3 +104,11 @@ func ValidateYesNo(value interface{}) (bool, string) {
 	}
 	return false, "must be yes or no"
 }
+
+// ValidateTrayAutostart checks tray-auto-start is used in macOS and its a bool
+func ValidateTrayAutostart(value interface{}) (bool, string) {
+	if runtime.GOOS == "darwin" {
+		return ValidateBool(value)
+	}
+	return false, "only supported in macOS"
```

Comment:
Since you added per-os files, you could also have 2 different definitions for `ValidateTrayAutostart` in these files rather than checking the OS at runtime.

Reply:
Moved this validation function also to the tray package